contrib/mpi-proxy-split/mpi-wrappers/mpi_wrappers.cpp

```diff
@@ -47,6 +48,7 @@ USER_DEFINED_WRAPPER(int, Init, (int *) argc, (char ***) argv) {
 }
 USER_DEFINED_WRAPPER(int, Init_thread, (int *) argc, (char ***) argv,
                      (int) required, (int *) provided) {
+  mana_state = RUNNING;
```

Comment:
So before the user application calls `MPI_Init` or `MPI_Init_thread`, the `mana_state` is `UNKNOWN_STATE`?
But we set `mana_state` to `RUNNING` before it actually calls the real `MPI_Init`?

In my mind, we either set `mana_state` to `RUNNING` as soon as the process started, or we should do it after the real `MPI_Init`.

It may don't matter for now, but I suggest we have a more concrete design.

Two more questions:
1. Are those two places the only places that set `mana_state` in the user thread? It may not have race logically, just want to confirm.
2. I guess it should not matter if the user checkpoint the application before it calls `MPI_Init`?

Reply:
I see. It should be my mistake initializing the variable. I will fix it today.

Reply:
I found the problem is the variable `mana_state` is defined in `mpi_plugin.cpp`, and declared in `mpi_plugin.h` as an extern variable. Usually `mpi_plugin.o` is compiled with `record-replay.o`. CI fails because the unit test `record-replay-comm-test.exe` only use `record-replay.o`, therefore the variable is only declared, but not defined. 

This fix update `mana_state` in different functions for each states (RUNNING, CKPT_COLLECTIVE, CKPT_P2P, etc.). Maybe a better way is using DMTCP event. This will be less intrusive. I will update this fix today.